content/kubermatic/master/advanced/addons/_index.en.md

```diff
@@ -115,78 +238,162 @@ spec:
+`quay.io/kubermatic/openshift-addons`, respectively.
 
-    Custom addon with manifest
+#### Creating a Docker Image
 
-   ```plaintext
-   .
-   ├── Dockerfile
-   └── foo
-       └── deployment.yaml
-   ```
+Depending on the cluster orchestrator, choose the Kubernetes or OpenShift Docker image as the basis for your
+own image. All addon manifest are stored in `/addons/<addonname>`, e.g. `/addons/kube-proxy`. When creating
+your own image, copy the manifests into a directory below `/addons`.
 
-    Dockerfile for custom addons:
+Suppose you have this directory structure:
 
-   ```dockerfile
-   FROM quay.io/kubermatic/addons:v0.0.1
+```plaintext
+.
+├─ Dockerfile
+└─ my-custom-addon
+   └─ deployment.yaml
+```
 
-   ADD ./ /addons/
-   ```
+A simple Dockerfile would be
 
-    Release the image with custom addon
+```dockerfile
+FROM quay.io/kubermatic/addons:YOUR_KUBERMATIC_VERSION
 
-   ```bash
-   export TAG=v1.0
-   docker build -t customer/addons:${TAG} .
-   docker push customer/addons:${TAG}
-   ```
+ADD ./ /addons/
+```
 
-1. Edit `values.yaml` you are using for the installation of Kubermatic. Change the path to the addons repository
+You can then build and push your customized addon image to a Docker registry of your choice:
 
-   ```yaml
-   kubermatic:
-     controller:
-       addons:
-         kubernetes
-           image:
-             repository: "quay.io/customer/addons" # <-- add your repo here
-   ```
+```bash
+export TAG=YOUR_KUBERMATIC_VERSION
+docker build -t customer/addons:${TAG} .
+docker push customer/addons:${TAG}
+```
 
-1. Add your addon to the list of default addons in `values.yaml`:
+#### Manifest Templating
 
-   ```yaml
-   kubermatic:
-     docker:
-     addons:
-       kubernetes
-         # list of addons to install into every user-cluster. All need to exist in the addons image
-         defaultAddons:
-         - foo # <-- add your addon here
-         - canal
-         - dashboard
-         - dns
-         - kube-proxy
-         - openvpn
-         - rbac
-   ```
+Kubermatic treats each file in an addon directory (`/addon/<addonname>`) as a
+[Go template](https://golang.org/pkg/text/template/), so it is possible to inject dynamic and
+user-cluster related information into addons at runtime. Please refer to the Go documentation for
+the exact templating syntax.
 
-1. Update the installation of Kubermatic
+Kubermatic injects the following variables into each file:
 
-   ```bash
-   helm upgrade --install --wait --timeout 300 --values values.yaml --namespace kubermatic kubermatic charts/kubermatic
-   ```
+* `.Kubeconfig` (string) - a YAML-encoded kubeconfig for accessing the user-cluster
+* `.MajorMinorVersion` (string) - Kubernetes/OpenShift version as `Major.Minor`
+* `.Cluster` (Cluster resource) - contains information about the user cluster, most often used
+  for the `.Cluster.Spec`
+* `.Credentials` (Credentials object) - contains cloud provider credentials
```

Comment:
How does the credentials object look like? Guess this is cloud specific? Where can the user find this?